modules/internal/ChapelArray.chpl

```diff
@@ -1831,7 +1831,7 @@ module ChapelArray {
         return false;
 
       // workaround for #21749:
-      } else if __primitive("call and fn resolves", "max", fullIdxType) {
+      } else if this.idxType != bool && __primitive("call and fn resolves", "max", fullIdxType) {
```

Comment:
maybe expand the comment above because now this line has 2 workarounds :)

Reply:
Now I know why I don't usually add comments!  :D

Reply:
Whether comments increase or decrease maintainability is definitely an interesting discussion :D

Reply:
Worst of all, I added this one without anyone even requesting me to!  Totally out of character... :)

Reply:
OK, there's a much bigger comment there now, for better or worse